### ISSUE TITLE
SIGSERVER while updating  a linode

### ISSUE THREAD

**Issue by lotooo** [linode/terraform-provider-linode#288]
### Terraform Version

```
$ terraform -v
Terraform v0.15.1
on linux_amd64
+ provider registry.terraform.io/hashicorp/aws v3.37.0
+ provider registry.terraform.io/hashicorp/null v3.1.0
+ provider registry.terraform.io/linode/linode v1.16.0
```

### Effected Resource(s)

- linode_10059931

### Terraform Configuration Files

Output of the plan : 

```
  # module.pubsub-dev-lhr.linode_instance.this["staging1-pubsub00"] will be updated in-place
  ~ resource "linode_instance" "this" {
      ~ boot_config_label = "staging-pubsub00.dev.lhr" -> "staging1-pubsub00_dev_lhr_plex_bz"
      ~ group             = "dev-london" -> "dev-lhr"
        id                = "10059931"
      ~ label             = "staging1-pubsub00_dev_lhr" -> "staging1-pubsub00_dev_lhr_plex_bz"
      ~ tags              = [
          + "dev-lhr",
          - "dev-london",
        ]
        # (11 unchanged attributes hidden)


      ~ config {
          ~ label        = "staging-pubsub00.dev.lhr" -> "staging1-pubsub00_dev_lhr_plex_bz"
            # (5 unchanged attributes hidden)


            # (2 unchanged blocks hidden)
        }


        # (4 unchanged blocks hidden)
    }
```

### Panic Output

```
$ terraform apply "/tmp/plan"
Acquiring state lock. This may take a few moments...
module.pubsub-dev-lhr.linode_instance.this["staging1-pubsub00"]: Modifying... [id=10059931]
╷
│ Error: Plugin did not respond
│ 
│   with module.pubsub-dev-lhr.linode_instance.this["staging1-pubsub00"],
│   on ../../modules/pubsub/main.tf line 30, in resource "linode_instance" "this":
│   30: resource "linode_instance" "this" {
│ 
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may contain more details.
╵
Releasing state lock. This may take a few moments...

Stack trace from the terraform-provider-linode_v1.16.0 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x16e7409]

goroutine 37 [running]:
github.com/linode/terraform-provider-linode/linode.resourceLinodeInstanceUpdate(0xc0001a3780, 0x1794ac0, 0xc0006b4600, 0x0, 0xffffffffffffffff)
        github.com/linode/terraform-provider-linode/linode/resource_linode_instance.go:983 +0x1c89
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).update(0xc0003a1680, 0x1d12038, 0xc0006f8a40, 0xc0001a3780, 0x1794ac0, 0xc0006b4600, 0x0, 0x0, 0x0)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.4.0/helper/schema/resource.go:302 +0x88
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc0003a1680, 0x1d12038, 0xc0006f8a40, 0xc0000bc1c0, 0xc000b4d0a0, 0x1794ac0, 0xc0006b4600, 0x0, 0x0, 0x0, ...)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.4.0/helper/schema/resource.go:404 +0x390
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc000136ee8, 0x1d12038, 0xc0006f8a40, 0xc0002405f0, 0xc0006f8a40, 0x1991820, 0xc00037eb00)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.4.0/helper/schema/grpc_provider.go:955 +0x8ef
github.com/hashicorp/terraform-plugin-go/tfprotov5/server.(*server).ApplyResourceChange(0xc000b4c3e0, 0x1d120e0, 0xc0006f8a40, 0xc0000bc000, 0xc000b4c3e0, 0xc00037eb40, 0xc0002d3ba0)
        github.com/hashicorp/terraform-plugin-go@v0.1.0/tfprotov5/server/server.go:331 +0xb5
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler(0x1991820, 0xc000b4c3e0, 0x1d120e0, 0xc00037eb40, 0xc00073e180, 0x0, 0x1d120e0, 0xc00037eb40, 0xc00001e000, 0x1510)
        github.com/hashicorp/terraform-plugin-go@v0.1.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:380 +0x214
google.golang.org/grpc.(*Server).processUnaryRPC(0xc00039f180, 0x1d24c58, 0xc000682d80, 0xc000316100, 0xc000a80540, 0x27241c0, 0x0, 0x0, 0x0)
        google.golang.org/grpc@v1.32.0/server.go:1194 +0x52b
google.golang.org/grpc.(*Server).handleStream(0xc00039f180, 0x1d24c58, 0xc000682d80, 0xc000316100, 0x0)
        google.golang.org/grpc@v1.32.0/server.go:1517 +0xd0c
google.golang.org/grpc.(*Server).serveStreams.func1.2(0xc00012a480, 0xc00039f180, 0x1d24c58, 0xc000682d80, 0xc000316100)
        google.golang.org/grpc@v1.32.0/server.go:859 +0xab
created by google.golang.org/grpc.(*Server).serveStreams.func1
        google.golang.org/grpc@v1.32.0/server.go:857 +0x1fd

Error: The terraform-provider-linode_v1.16.0 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

### Expected Behavior

Update works

### Actual Behavior

panic

**Comment by LBGarber** [comment]
Hello! 

It looks like this error is the result of your new instance label exceeding the 32 character limit, but this wasn't reported due to an issue with our instance update error handler. Setting your new label to a string less than 32 characters long should fix this issue.

A fix for the error handler should be in the next release. Let us know if you have any other issues! :smile:

**Comment by lotooo** [comment]
Thanks ! That worked !

**Comment by Charliekenney23** [comment]
The underlying cause of the panic while trying to surface the error has been resolved in the latest release, [v1.17.1](https://github.com/linode/terraform-provider-linode/releases/tag/v1.17.1).

Thanks for reporting this!
